domain/infosync/info.go

```diff
@@ -95,13 +95,38 @@ type InfoSyncer struct {
 // It will not be updated when tidb-server running. So please only put static information in ServerInfo struct.
 type ServerInfo struct {
 	ServerVersionInfo
-	ID             string `json:"ddl_id"`
-	IP             string `json:"ip"`
-	Port           uint   `json:"listening_port"`
-	StatusPort     uint   `json:"status_port"`
-	Lease          string `json:"lease"`
-	BinlogStatus   string `json:"binlog_status"`
-	StartTimestamp int64  `json:"start_timestamp"`
+	ID             string        `json:"ddl_id"`
+	IP             string        `json:"ip"`
+	Port           uint          `json:"listening_port"`
+	StatusPort     uint          `json:"status_port"`
+	Lease          string        `json:"lease"`
+	BinlogStatus   string        `json:"binlog_status"`
+	StartTimestamp int64         `json:"start_timestamp"`
+	ServerID       func() uint64 `json:"-"`
```

Comment:
why is it a function instead?

Reply:
To make `ServerID` in `ServerInfo` always the same as `Domain.serverID`. `Domain.serverID` would be changed when PD connection restored after broken.

Reply:
Will there be a problem when PD connection is restored but the info syncer has not yet reported the latest ServerID?

Reply:
I trigger a `StoreServerInfo()` [here](https://github.com/pingyu/tidb/blob/executor_global_kill/domain/domain.go#L1345), just after connection restored, to set the latest serverID to PD.